azext_iot/tests/helpers.py

```diff
@@ -127,6 +128,233 @@ def create_storage_account(
+    if create_namespace:
+        cmd(
+            "servicebus namespace create --name {} --resource-group {} --tags iotresource={}".format(
+                namespace_name, rg, resource_name
+            )
+        )
+
+    if create_queue:
+        cmd(
+            "servicebus queue create --namespace-name {} --resource-group {} --name {}".format(
+                namespace_name, rg, queue_name
+            )
+        )
+
+    if create_policy:
+        cmd(
+            "servicebus queue authorization-rule create --namespace-name {} --resource-group {} "
+            "--queue-name {} --name {} --rights Send".format(
+                namespace_name, rg, queue_name, policy_name
+            )
+        )
+
+    # Return a connection string, even if it was not created
+    queue_cs = cmd(
+        "servicebus queue authorization-rule keys list --namespace-name {} --resource-group {} "
+        "--queue-name {} --name {}".format(
+            namespace_name, rg, queue_name, policy_name
+        )
+    ).get_output_in_json()["primaryConnectionString"]
+
+    return queue_cs
+
+
+def create_cosmos_db(
+    cmd,
+    account_name: str,
+    database_name: str,
+    collection_name: str,
+    partition_key_path: str,
+    rg: str,
+    resource_name: str,
+    create_account: bool = True,
+    create_database: bool = True,
+    create_collection: bool = True,
+) -> str:
+    """
+    Create a cosmos db account, database, and collection (if needed) and return the cosmosdb connection string.
+    """
+
+    if create_account:
+        cmd(
+            'cosmosdb create --resource-group {} --name {} --tags iot_resource={}'.format(
+                rg, account_name, resource_name
+            )
+        )
+
+    if create_database:
+        cmd(
+            'cosmosdb sql database create --resource-group {} --account-name {} --name {}'.format(
+                rg, account_name, database_name
+            )
+        )
+
+    if create_collection:
+        cmd(
+            'cosmosdb sql container create --resource-group {} --account-name {} --database-name {} --name {} -p {}'.format(
+                rg, account_name, database_name, collection_name, partition_key_path
+            )
+        )
+
+    output = cmd(
+        'cosmosdb keys list --resource-group {} --name {} --type connection-strings'.format(rg, account_name)
+    ).get_output_in_json()
+
+    for cs_object in output["connectionStrings"]:
+        if cs_object["description"] == "Primary SQL Connection String":
+            return cs_object["connectionString"]
+
+
+def create_managed_identity(cmd, name: str, rg: str):
+    return cmd(
+        f"identity create -n {name} -g {rg}"
+    ).get_output_in_json()
+
+
+def get_role_assignments(cmd, scope, role):
+    return cmd(
+        'role assignment list --scope "{}" --role "{}"'.format(
+            scope, role
+        )
+    ).get_output_in_json()
+
+
+def assign_rbac_role(cmd, assignee: str, scope: str, role: str, max_tries: int = 10):
```

Comment:
This and get_role_assignments end up everywhere for tests. We should condense them into one function (prob a future task). Also another future task is to combine the other service creations in a neat way since they are shared btw dt, adu)

Reply:
killed the function